azure-mgmt-containerinstance/src/main/java/com/microsoft/azure/management/containerinstance/ContainerGroup.java

```diff
@@ -0,0 +1,729 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.containerinstance;
+
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.containerinstance.implementation.ContainerGroupInner;
+import com.microsoft.azure.management.containerinstance.implementation.ContainerInstanceManager;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.GroupableResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
+import com.microsoft.azure.management.resources.fluentcore.model.Attachable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import rx.Observable;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * An immutable client-side representation of an Azure Container Group.
+ */
+@Fluent
+@Beta(Beta.SinceVersion.V1_3_0)
+public interface ContainerGroup extends
+    GroupableResource<ContainerInstanceManager, ContainerGroupInner>,
+    Refreshable<ContainerGroup> {
+
+    /***********************************************************
+     * Getters
+     ***********************************************************/
+
+    /**
+     * @return the container instances in this container group
+     */
+    Map<String, Container> containers();
+
+    /**
+     * @return all the ports publicly exposed for this container group
+     */
+    Collection<Port> externalPorts();
+
+    /**
+     * @return the TCP ports publicly exposed for this container group
+     */
+    int[] externalTcpPorts();
+
+    /**
+     * @return the UDP ports publicly exposed for this container group
+     */
+    int[] externalUdpPorts();
+
+    /**
+     * @return the volumes for this container group
+     */
+    Map<String, Volume> volumes();
+
+    /**
+     * @return the Docker image registry servers by which the container group is created from
+     */
+    Collection<String> imageRegistryServers();
+
+    /**
+     * @return - `always` Always restart
+
+     */
+    String restartPolicy();
+
+    /**
+     * @return the IP address
+     */
+    String ipAddress();
+
+    /**
+     * @return - 'public' IP address is public
+     */
+    String ipAddressType();
+
+    /**
+     * @return the base level OS type required by the containers in the group
+     */
+    String osType();
```

Comment:
is this an arbitrary string or does it come from a predetermined set of known values?  (wouldn't this be an "enum"?)
especially since I see `OperatingSystemTypes` extensible enum auto-gen'd later in this PR.

Reply:
Comes from a predefined set of values. I changed the return to reflect that; btw previously I returned the value of it as a string because I figured the resource is read only and strings are much easy to work with...